### DISCUSSION TITLE
MenuItem ICommand.CanExecuteChanged does not work via styles?

### DISCUSSION THREAD

**Discussion by Tokter** [AvaloniaUI/Avalonia#11110] [General]
I'm new to Avalonuia, so I'm not sure if this is a bug, or if I'm missing something.

If I'm binding a MenuItem to a command like this:

```xml
<MenuItem Header="Test" Command="{Binding	SaveCommand}" />
```

Then everything works fine, I can call RaiseCanExecuteChanged on my command and the menu item enables/disables accordingly:

```C#
        public void RaiseCanExecuteChanged()
        {
            CanExecuteChanged?.Invoke(this, EventArgs.Empty);
        }
```

If I bind the same command to a dynamically created MenuItem, and do the bindings via style. Then RaiseCanExecuteChanged() does nothing, the only time CanExecute() gets checked, is the first time the menu appears.

```XML
		<Style Selector="MenuItem">
			<Setter Property="Header" Value="{Binding Name}"/>
			<Setter Property="Items" Value="{Binding Items}"/>
			<Setter Property="Command" Value="{Binding Command}"/>
      <Setter Property="CommandParameter" Value="{Binding Playground}"/>
			<Setter Property="Icon">
				<Template>
					<PathIcon Data="{Binding Icon}" Width="14" Height="14" />
				</Template>
			</Setter>
		</Style>
````

Any help is appreciated, thanks!

**Comment by timunie** [comment]
Please share a minimal sample to reproduce, so we see all code including your command definition. Also often using CompiledBindings does help a lot to debug binding related issues.

**Comment by Tokter** [comment]
Here's a minimal project:
[AvaloniaMenuItemIssue.zip](https://github.com/AvaloniaUI/Avalonia/files/11307646/AvaloniaMenuItemIssue.zip)

Steps to see the issue:
- First open both menus once
- Execute command 1, that should disable command 1 and enable command 2
- Inspect menus:

Compile time menu is in sync with the buttons:
![image](https://user-images.githubusercontent.com/3439101/233925230-1f3a7c27-d971-44df-99e3-d16bd383f344.png)

Runtime bound menu has still Command 1 as active:
![image](https://user-images.githubusercontent.com/3439101/233925357-1f436ae4-b145-44ae-b926-0c612555c7b0.png)

**Reply by timunie** [reply]
I can reproduce it using your sample, also using latest nightly build. So I assume it is a bug. Will convert it into a bug report therefore. 

Thank you for the great sample 💐

**Comment by maxkatz6** [comment]
It is by design. At least in 0.10.
Command is a direct property and did not make much sense to be styleable. I.e., styles are designed to style the control, not to setup default bindings from the data context, which might or might not actually exist (as style can be defined anywhere and applied on any menu item).

In 11.0 previews old behavior is even stricter with direct properties (exception), but many properties including Command now are styleable, so your code is allowed and works in 11.0 previews.
See long discussions in that ticket and referenced ones: https://github.com/AvaloniaUI/Avalonia/issues/9944

**Reply by maxkatz6** [reply]
For a reason I thought it's an issue for 0.10 version, but apparently it' still reproducible in 11.0. Reopened issue.
